sbirez/auth_utils.py

```diff
@@ -0,0 +1,104 @@
+from django.contrib.auth import get_user_model
+from django.conf import settings
+from django.contrib.auth.forms import PasswordResetForm, SetPasswordForm
+try:
+    from django.utils.http import urlsafe_base64_decode as uid_decoder
+except:
+    # make compatible with django 1.5
+    from django.utils.http import base36_to_int as uid_decoder
+from django.contrib.auth.tokens import default_token_generator
+from django.contrib.auth.hashers import identify_hasher
+from rest_framework import serializers
+from rest_framework.exceptions import ValidationError
+import rest_auth.serializers
+from datetime import timedelta
+from django.utils import timezone
+from sbirez.models import PasswordHistory, SbirezUser
+from Levenshtein import distance
+
+def password_check(value, user):
+    """
+    Check that the password passes the following validation:
+
+    Password is long enough (based on configuration setting)
+    Password contains at least one digit
+    Password contains at least one letter
+    Password contains at least one uppercase character
+    Password contains at least one special character
+    Password has not been reused in a configurable number of password cycles
+    """
+    if len(value) < settings.AUTH_PASSWORD_LENGTH:
+        raise serializers.ValidationError('Password is too short')
+    # check for digit
+    if not any(char.isdigit() for char in value):
+        raise serializers.ValidationError('Password must contain at least 1 digit.')
+    # check for letter
+    if not any(char.isalpha() for char in value):
+        raise serializers.ValidationError('Password must contain at least 1 letter.')
+    # check for upper case letter
+    if not any(char.isupper() for char in value):
+        raise serializers.ValidationError('Password must contain at least 1 uppercase letter.')
+    # check for symbol
+    predef = set('!@#$%&*()')
```

Comment:
If we don't want specifically to limit it to these characters (which exclude, for instance, []{}:;, etc), could:

    if not any(not char.isalnum() for char in value)